Dragonflight/PriestDiscipline.lua

```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
         end,
     },
 
-    -- Smites an enemy for 1,315 Holy damage.
+    shadowfiend = {
+        id = 34433,
+        cast = 0,
+        cooldown = 180,
+        gcd = "spell",
+        school = "shadow",
+        talent = "shadowfiend",
+        notalent = "mindbender",
+        startsCombat = true,
+        texture = 136199,
+        toggle = "cooldowns",
+
+        handler = function ()
+            summonPet( "shadowfiend", 15 )
+            applyBuff( "shadowfiend" )
+        end,
+    },
+
+
+    shadow_word_death = {
+        id = 32379,
+        cast = 0,
+        charges = 1,
+        cooldown = 8,
+        recharge = 8,
+        gcd = "spell",
+        school = "shadow",
+        damage = 1,
+
+        spend = 0.005,
+        spendType = "mana",
+
+        talent = "shadow_word_death",
+        startsCombat = true,
+
+        usable = function ()
+            if settings.sw_death_protection == 0 then return true end
+            return health.percent >= settings.sw_death_protection, "health percent [ " .. health.percent .. " ] is below user setting [ " .. settings.sw_death_protection .. " ]"
+        end,
+
+        handler = function ()
+            if talent.death_and_madness.enabled then
+                applyDebuff( "target", "death_and_madness_debuff" )
+            end
+
+            if talent.inescapable_torment.enabled then
+                if buff.mindbender.up then buff.mindbender.expires = buff.mindbender.expires + (talent.inescapable_torment.rank * 0.5)
+                elseif buff.shadowfiend.up then buff.shadowfiend.expires = buff.shadowfiend.expires + (talent.inescapable_torment.rank * 0.5) end
+            end
+
+            if talent.expiation.enabled then
+                if talent.purge_the_wicked.enabled then
+                    if debuff.purge_the_wicked.remains <= 6 then
+                        removeDebuff( "purge_the_wicked")
+                    else
+                        debuff.purge_the_wicked.expires = debuff.purge_the_wicked.expires - 6
+                    end
+                else
+                    if debuff.shadow_word_pain.remains <= 6 then
+                        removeDebuff( "shadow_word_pain")
+                    else
+                        debuff.shadow_word_pain.expires = debuff.shadow_word_pain.expires - 6
+                    end
+                end
+            end
+        end,
+    },
+
+
     smite = {
         id = 585,
-        cast = 1.5,
+        cast = function() return 1.5*haste end,
```

Comment:
This should already be hasted automatically.

Try `/dump Hekili.Class.abilities.smite.cast` in-game to verify before making this change.

Reply:
You're correct!